cypress/integration/Dialog.spec.ts

```diff
@@ -1,11 +1,198 @@
+        if (action === 'realTouch') {
+          cy.findByLabelText('count up').realClick();
+        } else {
+          cy.findByLabelText('count up').realTouch();
+        }
+        cy.findByLabelText('count up').should('have.text', count);
+      });
+  }
+  function shouldAllowOutsideInteraction(action: 'realClick' | 'realTouch') {
+    cy.findByLabelText('count up')
+      .invoke('text')
+      .then((count) => {
+        if (action === 'realTouch') {
+          cy.findByLabelText('count up').realClick();
+        } else {
+          cy.findByLabelText('count up').realTouch();
+        }
+        cy.findByLabelText('count up').should('not.have.text', count);
+      });
+  }
+
+  describe('given a modal dialog', () => {
+    it('can be open/closed with a keyboard', () => {
+      // using keyboard on open/close buttons
+      cy.findByText('open').focus();
+      cy.realPress('Space');
+      shouldBeOpen();
+      cy.findByText('close').should('be.focused');
+      cy.realPress('Space');
+      shouldBeClosed();
+      cy.findByText('open').should('be.focused');
+
+      // using keyboard on open button and close with escape
+      cy.realPress('Space');
+      shouldBeOpen();
+      cy.realPress('Escape');
+      shouldBeClosed();
+    });
+
+    it('can be open/closed with a pointer', () => {
+      // using pointer on open/close buttons
+      cy.findByText('open').click();
+      shouldBeOpen();
+      cy.findByText('close').should('be.focused').click();
+      shouldBeClosed();
+      cy.findByText('open').should('be.focused');
+
+      // using mouse inside dialog, then on a button outside
+      cy.findByText('open').click();
+      shouldBeOpen();
+      cy.findByText('title').click();
+      shouldBeOpen();
+      shouldNotAllowOutsideInteraction('realClick');
+      shouldBeClosed();
+
+      // using touch on a button outside
+      cy.findByText('open').click();
+      shouldBeOpen();
+      shouldNotAllowOutsideInteraction('realTouch');
+      shouldBeClosed();
+
+      // using mouse on an input outside
+      cy.findByText('open').click();
+      shouldBeOpen();
+      cy.findByPlaceholderText('name').realClick();
+      cy.findByPlaceholderText('name').should('not.be.focused');
+      shouldBeClosed();
+
+      // using touch on an input outside
+      cy.findByText('open').click();
+      shouldBeOpen();
+      cy.findByPlaceholderText('name').realTouch();
+      cy.findByPlaceholderText('name').should('not.be.focused');
+      shouldBeClosed();
+
+      // turn on animation
+      cy.findByLabelText('animated').click();
+
+      // using mouse on an input outside an animated dialog
+      cy.findByText('open').click();
+      shouldBeOpen();
+      cy.findByPlaceholderText('name').realClick();
+      cy.findByPlaceholderText('name').should('not.be.focused');
+      shouldBeClosed();
+
+      // finally, make sure that we have no pointer-events regression
+      shouldAllowOutsideInteraction('realClick');
```

Comment:
```suggestion
      // finally, ensure that pointer-events have been reset and interactions restored
      shouldAllowOutsideInteraction('realClick');
```

Should we be clearer here? is this just to ensure that `pointer-events: none` is no longer set on the body?

Also, is there any chance of a footgun when the counter is incremented as part of this assertion?

Reply:
> is this just to ensure that pointer-events: none is no longer set on the body?

Yes.

> Also, is there any chance of a footgun when the counter is incremented as part of this assertion?

Sorry, what do you mean?

Reply:
> Sorry, what do you mean?

I was just wondering whether we need to increment again given the context, or if leaning on Cypress to tell us that `pointer-events: none` was still set (via a normal `.click`) would be clearer.

No biggie tbh, in some ways testing the cause and effect with the increment is better.

Reply:
Yeah I feel like testing the actual result is stronger.